backend/functions/tasks.py

```diff
@@ -204,4 +203,161 @@ def sentence_text_translate_and_save_translation_pairs(
+
+    Args:
+        languages (list): List of output languages for the translations.
+        input_dataset_instance_id (int): ID of the input dataset instance.
+        output_dataset_instance_id (int): ID of the output dataset instance.
+        batch_size (int): Number of sentences to be translated in a single batch.
+        api_type (str): Type of API to be used for translation. (default: indic-trans)
+            Allowed - [indic-trans, google]
+        checks_for_particular_languages (bool): If True, checks for the particular languages in the translations.
+    """
+
+    # Get the output dataset instance
+    output_dataset_instance = dataset_models.DatasetInstance.objects.get(
+        instance_id=output_dataset_instance_id
+    )
+
+    # Collect all the Conversation dataitems for the input DatasetInstance and convert to dataframe
+    conversation_dataitems = dataset_models.Conversation.objects.filter(
+        instance_id=input_dataset_instance_id
+    ).values_list("id", "scenario", "prompt", "conversation_json", "language")
+
+    conversation_dataitems_df = pd.DataFrame(
+        conversation_dataitems,
+        columns=["id", "scenario", "prompt", "conversation_json", "language"],
+    )
+
+    # Check if the dataframe is empty
+    if conversation_dataitems_df.shape[0] == 0:
+
+        # Update the task status
+        self.update_state(
+            state="FAILURE",
+            meta={
+                "No sentences to upload.",
+            },
+        )
+
+        raise Exception("The conversation data is empty.")
+
+    # Iterate through the languages
+    for output_language in languages:
+
+        all_translated_conversation_objects = []
+        # Iterate through the conversation dataitems
+        for index, row in conversation_dataitems_df.iterrows():
+
+            # Get the instance of the Conversation dataitem
+            conversation_dataitem = dataset_models.Conversation.objects.get(
+                id=row["id"]
+            )
+
+            # Get the conversation JSON and iterate through it
+            conversation_json = row["conversation_json"]
+            translated_conversation_json = []
+            for conversation in conversation_json:
+
+                # Get the sentence list, scenario and prompt
+                sentences_to_translate = dict(conversation).get("sentences", [])
+                speaker_id = dict(conversation).get("speaker_id")
+                sentence_count = len(sentences_to_translate)
+                sentences_to_translate.append(row["scenario"])
+                sentences_to_translate.append(row["prompt"])
+
+                # Check the API type
+                if api_type == "indic-trans":
+
+                    # Get the translation using the Indictrans NMT API
+                    translations_output = get_batch_translations_using_indictrans_nmt_api(
+                        sentence_list=sentences_to_translate,

```

Comment:
Since we will be using Machine Translation in multiple places in the code, shall we create a util function called `get_batch_translations()`, and also pass the `api_type` to that? Let that function itself return an exception if `api_type` is not supported.

That way, we can keep the code cleaner. For example, in the next sprint, we will be adding support for Microsoft Translate Service into Shoonya; it will be helpful in that case to make minimal changes.